pkg/user/user_integration_test.go

```diff
@@ -84,20 +76,102 @@ func TestUserHandler(t *testing.T) {
-		require.NoError(t, err)
-		err = userService.ValidateEmail(u2.EmailToken)
+		createUser(t, client, userService, "user3@dhis2.org", "oneoneoneoneoneoneone111")
+	}
+
+	t.Run("SignOut", func(t *testing.T) {
+		t.Parallel()
+
+		expires, err := time.Parse(time.RFC3339, "0001-01-01T00:00:00Z")
 		require.NoError(t, err)
 
-		var user3 model.User
-		client.PostJSON(t, "/users", strings.NewReader(`{
-			"email":    "user3@dhis2.org",
-			"password": "oneoneoneoneoneoneone111"
-		}`), &user3)
+		expectedCookies := []*http.Cookie{
+			{
+				Name:       "accessToken",
+				Value:      "",
+				Path:       "/",
+				Domain:     "",
+				Expires:    expires,
+				RawExpires: "",
+				MaxAge:     -1,
+				Secure:     true,
+				HttpOnly:   true,
+				SameSite:   0,
+				Raw:        "accessToken=; Path=/; Max-Age=0; HttpOnly; Secure",
+				Unparsed:   nil,
+			},
+			{
+				Name:       "refreshToken",
+				Value:      "",
+				Path:       "/",
+				Domain:     "",
+				Expires:    expires,
+				RawExpires: "",
+				MaxAge:     -1,
+				Secure:     true,
+				HttpOnly:   true,
+				SameSite:   0,
+				Raw:        "refreshToken=; Path=/; Max-Age=0; HttpOnly; Secure",
+				Unparsed:   nil,
+			},
+			{
+				Name:       "rememberMe",
+				Value:      "",
+				Path:       "/",
+				Domain:     "",
+				Expires:    expires,
+				RawExpires: "",
+				MaxAge:     -1,
+				Secure:     true,
+				HttpOnly:   true,
+				SameSite:   0,
+				Raw:        "rememberMe=; Path=/; Max-Age=0; HttpOnly; Secure",
+				Unparsed:   nil,
+			},
+		}
 
-		require.Equal(t, "user3@dhis2.org", user3.Email)
-		require.Empty(t, user3.Password)
-	}
+		t.Run("NoToken", func(t *testing.T) {
+			request := client.NewRequest(t, http.MethodDelete, "/users", nil)
+
+			response, err := client.Client.Do(request)
+			require.NoError(t, err)
+
+			require.Equal(t, http.StatusUnauthorized, response.StatusCode)
+			assert.EqualValues(t, expectedCookies, response.Cookies())
+		})
+
+		t.Run("ValidToken", func(t *testing.T) {
+			createUser(t, client, userService, "user4@dhis2.org", "oneoneoneoneoneoneone111")
+
+			var tokens *token.Tokens
+			client.PostJSON(t, "/tokens", strings.NewReader(`{}`), &tokens, inttest.WithBasicAuth("user4@dhis2.org", "oneoneoneoneoneoneone111"))
+
+			request := client.NewRequest(t, http.MethodDelete, "/users", nil, inttest.WithAuthToken(tokens.AccessToken))
+
+			response, err := client.Client.Do(request)
+			require.NoError(t, err)
+
+			require.Equal(t, http.StatusOK, response.StatusCode)
+			assert.EqualValues(t, expectedCookies, response.Cookies())
+		})
+
+		t.Run("ExpiredToken", func(t *testing.T) {
+			createUser(t, client, userService, "user5@dhis2.org", "oneoneoneoneoneoneone111")
+
+			var tokens *token.Tokens
+			client.PostJSON(t, "/tokens", strings.NewReader(`{}`), &tokens, inttest.WithBasicAuth("user5@dhis2.org", "oneoneoneoneoneoneone111"))
+
+			time.Sleep(time.Duration(tokens.ExpiresIn) * time.Second)
```

Comment:
Are we now waiting 10 seconds? Maybe we can shrink that config for the test. I think https://pkg.go.dev/time#After would be better so the runtime gets a chance to maybe run another test while we wait here.

`<-time.After(time.Duration(tokens.ExpiresIn) * time.Second)`